create-custom-styles/src/main/java/com/esri/samples/custom_style/CustomStyleSample.java

```diff
@@ -0,0 +1,188 @@
+
+import com.esri.arcgisruntime.concurrent.ListenableFuture;
+import com.esri.arcgisruntime.data.Geodatabase;
+import com.esri.arcgisruntime.geometry.Point;
+import com.esri.arcgisruntime.layers.FeatureLayer;
+import com.esri.arcgisruntime.layers.OpenStreetMapLayer;
+import com.esri.arcgisruntime.loadable.LoadStatus;
+import com.esri.arcgisruntime.mapping.Basemap;
+import com.esri.arcgisruntime.mapping.ArcGISMap;
+import com.esri.arcgisruntime.mapping.view.MapView;
+import com.esri.arcgisruntime.symbology.*;
+import javafx.application.Application;
+import javafx.scene.Scene;
+import javafx.scene.layout.StackPane;
+import javafx.stage.Stage;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.concurrent.ExecutionException;
+
+public class CustomStyleSample extends Application {
+
+  private MapView mapView;
+  private ArcGISMap map;
+
+  public static void main(String[] args) {
+
+    Application.launch(args);
+  }
+
+  @Override
+  public void start(Stage stage) {
+
+    // set the title and size of the stage and show it
+    stage.setTitle("Custom Style App");
+    stage.setWidth(800);
+    stage.setHeight(700);
+    stage.show();
+
+    // create a JavaFX scene with a stack pane as the root node and add it to the scene
+    StackPane stackPane = new StackPane();
+    Scene scene = new Scene(stackPane);
+    stage.setScene(scene);
+
+    // create a MapView to display the map and add it to the stack pane
+    mapView = new MapView();
+    stackPane.getChildren().add(mapView);
+
+    // create an ArcGISMap with an Open Street Map Basemap
+    OpenStreetMapLayer openStreetMapLayer = new OpenStreetMapLayer();
+    map = new ArcGISMap();
+    Basemap basemap = new Basemap();
+    basemap.getBaseLayers().add(openStreetMapLayer);
+    map.setBasemap(basemap);
+
+    // load the style and make a renderer
+    UniqueValueRenderer renderer = MakeRendererFromStyle();
+
+    // add the data from mobile geodatabase using the renderer
+    addMobileGDBData(renderer);
+
+    // display the map by setting the map on the map view
+    mapView.setMap(map);
+
+    // set an initial viewpoint
+    Point point = new Point(-302209.076759, 7568136.965982);
+    mapView.setViewpointCenterAsync(point, 5000);
+
+  }
+
+  /**
+   * Method to load a style file and use the symbols to create a renderer
+   *
+   * @return renderer using symbols in style file
+   */
+  private UniqueValueRenderer MakeRendererFromStyle() {
+    // create the unique value renderer
+    UniqueValueRenderer uniqueValueRenderer = new UniqueValueRenderer();
+    uniqueValueRenderer.getFieldNames().add("InsectGroup");
+
+    // default symbol
+    SimpleMarkerSymbol simpleMarkerSymbol = new SimpleMarkerSymbol(SimpleMarkerSymbol.Style.TRIANGLE, 0xFFFF0000,10);
+    uniqueValueRenderer.setDefaultSymbol(simpleMarkerSymbol);
+
+    // load the style
+    SymbolStyle symbolStyle = new SymbolStyle("data/Insects.stylx");
```

Comment:
I think `symbolStyle` needs to be a field to protect it from garbage collection before its DoneLoading listener is called

Reply:
There's now a field called `symbolStyle`, but this is declaring and using a local variable `symbolStyle`